app/src/main/java/io/token/sample/Application.java

```diff
@@ -0,0 +1,281 @@
+package io.token.sample;
+
+import static com.google.common.base.Charsets.UTF_8;
+import static io.grpc.Status.Code.NOT_FOUND;
+import static io.token.TokenClient.TokenCluster.SANDBOX;
+import static io.token.proto.common.alias.AliasProtos.Alias.Type.EMAIL;
+import static io.token.util.Util.generateNonce;
+
+import com.google.common.io.Resources;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import io.grpc.StatusRuntimeException;
+import io.token.proto.common.account.AccountProtos.BankAccount;
+import io.token.proto.common.account.AccountProtos.BankAccount.Domestic;
+import io.token.proto.common.alias.AliasProtos.Alias;
+import io.token.proto.common.member.MemberProtos.Profile;
+import io.token.security.UnsecuredFileSystemKeyStore;
+import io.token.tokenrequest.TokenRequest;
+import io.token.tpp.Member;
+import io.token.tpp.Representable;
+import io.token.tpp.TokenClient;
+import io.token.tpp.tokenrequest.TokenRequestCallback;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Map;
+
+import spark.Response;
+import spark.Spark;
+
+/**
+ * Application main entry point.
+ */
+public class Application {
+    private static final String CSRF_TOKEN_KEY = "csrf_token";
+    private static final TokenClient tokenClient = initializeSDK();
+    private static final Member cbpiiMember = initializeMember(tokenClient);
+    private static final Double AMOUNT = 10.0;
+    private static final String CURRENCY = "USD";
+
+    /**
+     * Main function.
+     *
+     * @param args command line arguments
+     * @throws IOException thrown on errors
+     */
+    public static void main(String[] args) throws IOException {
+        // Initializes the server
+        Spark.port(3000);
+
+        // (redirect flow) Endpoint for requesting access to account balances
+        Spark.get("/request-funds-confirmation", (req, res) -> {
+            String callbackUrl = req.scheme() + "://" + req.host() + "/confirm-funds";
+            String tokenRequestUrl = getTokenRequestUrl(callbackUrl, res);
+
+            //send a 302 redirect
+            res.status(302);
+            res.redirect(tokenRequestUrl);
+            return null;
+        });
+        // (redirect flow) Endpoint for transfer payment, called by client side after user approves payment.
+        Spark.get("/confirm-funds", (req, res) -> {
+            String callbackUrl = req.url() + "?" + req.queryString();
+
+            // retrieve CSRF token from browser cookie
+            String csrfToken = req.cookie(CSRF_TOKEN_KEY);
```

Comment:
Not sure why? what do you suggest?

Reply:
Because that's how a CSRF attack works (https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)#Examples)

Cookies are automatically sent by the browser regardless of where the request originates from.
https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)#Using_a_secret_cookie

The only way to prevent this default behavior is to use the SameSite directive.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Directives

Which I don't think we are using and it isn't compatible with IE.

There are many ways to fix this, but it requires a bigger conversation.

Reply:
I understand, I was asking about this case specifically.
I am not sure if you know how the csrf token is used in the token request flow, so I suggest you read this first: https://developer.token.io/docs/#flow-overview
This is what the doc suggests:
```
Note: Sensitive strings (such as sessionCookie) can be used directly as the csrfToken in the Token SDK since it hashes the token before using it. For more details on CSRF attacks against OAuth 2.0 and mitigation techniques, refer to this rfc.

If your application does not have user authentication, you can use Util.generateNonce to generate a secure random string to use as a CSRF token.
```
If you think there is a concern, happy to discuss the alternatives in person.

Reply:
Let's just talk about this in person. Too much back and forth :D